Assets/MixedRealityToolkit-SDK/Features/UX/Interactable/Scripts/States/States.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX
+{
+    [CreateAssetMenu(fileName = "States", menuName = "Interactable/State", order = 1)]
+    public class States : ScriptableObject
+    {
+        public List<State> StateList;
+        public int DefaultIndex = 0;
+        public Type StateType;
+        public string[] StateOptions;
+        public Type[] StateTypes;
+        public string StateLogicName = "InteractableStates";
+
+        //!!! finish making states work, they shoulg initiate the type and run the logic during play mode.
+        private void OnEnable()
+        {
+            SetupStateOptions();
+        }
+        
+        public State[] GetStates()
+        {
+            return StateList.ToArray();
+        }
+
+        public InteractableStates SetupLogic()
+        {
+            int index = ReverseLookup(StateLogicName, StateOptions);
+            StateType = StateTypes[index];
+            InteractableStates stateLogic = (InteractableStates)Activator.CreateInstance(StateType, StateList[DefaultIndex]);
+            List<State> stateListCopy = new List<State>();
+            for (int i = 0; i < StateList.Count; i++)
+            {
+                State state = new State();
+                state.ActiveIndex = StateList[i].ActiveIndex;
+                state.Bit = StateList[i].Bit;
+                state.Index = StateList[i].Index;
+                state.Name = StateList[i].Name;
+                state.Value = StateList[i].Value;
+                stateListCopy.Add(state);
+            }
+            stateLogic.ImportStates(stateListCopy);
+            
+            return stateLogic;
+        }
+
+        public void SetupStateOptions()
+        {
+            List<Type> stateTypes = new List<Type>();
+            List<string> names = new List<string>();
+
+            Assembly assembly = typeof(InteractableStates).GetTypeInfo().Assembly;
+            foreach (Type type in assembly.GetTypes())
+            {
+                TypeInfo info = type.GetTypeInfo();
+                if (info.BaseType.Equals(typeof(InteractableStates)) || type.Equals(typeof(InteractableStates)))
+                {
+                    stateTypes.Add(type);
+                    names.Add(type.Name);
+                }
+            }
+
+            /* works with IL2CPP but not in .NET
```

Comment:
Is this code being saved for something else? If you're about to revisit it in your next PR, add a //TODO comment explaining what needs to be done, otherwise best to delete it.

Reply:
This is the code change that fixed the errors building for .NET. The older code felt more robust, especially type.IsSubclassOf(typeof(InteractableStates)). Right now I'm only a level deep so (info.BaseType.Equals(typeof(InteractableStates)) || type.Equals(typeof(InteractableStates))) is working fine, but if things are extended more should I do a for loop until info.BaseType == null? I'm willing to delete it though, just wanted more testing and feedback first.